docs/source/guides/feature_descriptions.rst

```diff
@@ -0,0 +1,116 @@
+Generating Feature Descriptions
+================================
+As features become more complicated, their names can become harder to understand. Both the :func:`featuretools.describe_feature` function and the :func:`featuretools.graph_feature` function can help explain what a feature is and the steps Featuretools took to generate it. Additionally, the ``describe_feature`` function can be augmented by providing custom definitions and templates to improve the resulting descriptions. 
+
+.. ipython:: python
+    :suppress:
+
+    import featuretools as ft
+    es = ft.demo.load_mock_customer(return_entityset=True)
+
+    feature_defs = ft.dfs(entityset=es,
+                          target_entity="customers",
+                          agg_primitives=["mean", "sum", "mode", "n_most_common"],
+                          trans_primitives=["month", "hour"],
+                          max_depth=2,
+                          features_only=True)
+
+By default, ``describe_feature`` uses the existing variable and entity names and the default primitive description templates to generate feature descriptions. 
+
+.. ipython:: python
+
+    feature_defs[7]
+    ft.describe_feature(feature_defs[7])
+
+.. ipython:: python
+
+    feature_defs[12]
+    ft.describe_feature(feature_defs[12])
+
+Improving Descriptions
+~~~~~~~~~~~~~~~~~~~~~~~
+While the default descriptions can be helpful, they can also be further improved by providing custom definitions of variables and features, and by providing alternative templates for primitive descriptions. 
+
+Feature Descriptions
+---------------------
+Custom feature definitions will get used in the description in place of the automatically generated description. This can be used to better explain what a variable or feature is, or to provide descriptions that take advantage of a user's existing knowledge about the data or domain. 
+
+.. ipython:: python
+
+    feature_descriptions = {
+        'customers: join_date': 'the date the customer joined'}
+
+    ft.describe_feature(feature_defs[8], feature_descriptions=feature_descriptions)
```

Comment:
for the continuity angle, I don't think we've feature_defs[8] before this block so printing it out could be helpful